tests/build-tests.js

```diff
@@ -1,174 +1,203 @@
+                                    'dom1.com' // duplicate should be removed
+                                ],
+                                reason: 'Reason1.'
+                            }
+                        ]
+                    },
+                    'f4.com': {
+                        rules: [
+                            {
+                                rule: 'f4.com/foo',
+                                domains: [
+                                    'dom1.com'
+                                ],
+                                reason: 'Reason1.'
+                            },
+                            {
+                                rule: 'f4.com/foo.bar', // more specific, so should get put earler
+                                domains: [
+                                    'dom1.com'
+                                ],
+                                reason: 'Reason1.'
+                            }
+                        ]
+                    }
+                }
             }
-        ]
-    },
-    'f2.com': { // in ta1 not ta2
-        rules: [
-            {
-                rule: 'f2.com/foo',
-                domains: [
-                    'dom1.com',
-                    'dom0.com', // should get moved to start (lexical ordering)
-                    'dom1.com' // duplicate should be removed
-                ],
-                reason: 'Reason1.'
-            }
-        ]
-    },
-    'f4.com': {
-        rules: [
-            {
-                rule: 'f4.com/foo',
-                domains: [
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            },
-            {
-                rule: 'f4.com/foo.bar', // more specific, so should get put earler
-                domains: [
-                    'dom1.com'
-                ],
-                reason: 'Reason1.'
-            }
-        ]
+        }
     }
 }
 
 const ta2 = {
-    'f1.com': {
-        rules: [
-            {
-                rule: 'www.f1.com/foo/bar',
-                domains: [
-                    'dom1.com', // overlaps
-                    'dom3.net', // new domain
-                    'dom0.com' // should get moved to start (lexical ordering)
-                ],
-                reason: 'Reason2.'
-            },
-            {
-                rule: 'www.f1.com/foo', // new rule
-                domains: [
-                    'dom1.com',
-                    'dom0.com' // should be sorted in output
-                ],
-                reason: 'Reason1.'
-            }
-        ]
-    },
-    'f3.com': { // in ta2 not ta1
-        rules: [
-            {
-                rule: 'f3.com/foo',
-                domains: [
-                    'dom1.com',
-                    'dom0.com', // should be sorted in output
-                    'dom1.com' // duplicate should be removed
-                ],
-                reason: 'Reason1.'
+    features: {
+        trackerAllowlist: {
```

Comment:
why are we testing only trackerAllowlist? Is this a good representation of all possible features? e.g. it doesn't seem to have regular exceptions only settings/exceptions

Reply:
There are other settings tests in config-tests. This was modified from Ben's code to test tracker allowlist merging but refactored to use the deep merge method added here.